src/main/kotlin/me/serce/solidity/lang/types/inference.kt

```diff
@@ -37,25 +37,8 @@ fun getSolType(type: SolTypeName?): SolType {
         }
       }
     }
-    is SolUserDefinedTypeName -> {
-      val name = type.name
-      if (name != null && isInternal(name)) {
-        val internalType = SolInternalTypeFactory.of(type.project).byName(name)
-        return internalType ?: SolUnknown
-      }
-      val resolvedTypes = SolResolver.resolveTypeNameUsingImports(type)
-      return resolvedTypes.asSequence()
-        .map {
-          when (it) {
-            is SolContractDefinition -> SolContract(it)
-            is SolStructDefinition -> SolStruct(it)
-            is SolEnumDefinition -> SolEnum(it)
-            else -> null
-          }
-        }
-        .filterNotNull()
-        .firstOrElse(SolUnknown)
-    }
+    is SolUserDefinedLocationTypeName -> getSolTypeFromUserDefinedTypeName(type.userDefinedTypeName!!)
```

Comment:
I think in some cases on the code which is not fully correct `userDefinedTypeName` might return `null`. WDYT of doing something like the following? 
```kotlin
is SolUserDefinedLocationTypeName -> 
  type.userDefinedTypeName?.let {name -> getSolTypeFromUserDefinedTypeName(name) }
  ?: SolUnknown
```

Reply:
Sure, ok